src/main/java/software/amazon/smithy/lsp/SmithyLanguageServer.java

```diff
@@ -718,13 +680,17 @@ public CompletableFuture<CompletionItem> resolveCompletionItem(CompletionItem un
         LOGGER.finest("Definition");
 
         String uri = params.getTextDocument().getUri();
-        if (!projects.isTracked(uri)) {
+        ProjectAndFile projectAndFile = state.findProjectAndFile(uri);
+        if (projectAndFile == null) {
             client.unknownFileError(uri, "definition");
-            return completedFuture(Either.forLeft(Collections.emptyList()));
+            return completedFuture(null);
+        }
+
+        if (!(projectAndFile.file() instanceof SmithyFile smithyFile)) {
```

Comment:
Nit: 3 repetitions of this logic. I wonder if we could consolidate

Reply:
In the future, there will actually be different paths for BuildFiles. We will have to refactor it then anyways, but with the knowledge of how it needs to end up working